media-proxy/src/proxy_context.cc

```diff
@@ -664,6 +665,7 @@ int ProxyContext::RxStart_mtl(const mcm_conn_param *request)
     std::cout << "session id: " << st_ctx->id << std::endl;
     INFO("%s, session id: %d", __func__, st_ctx->id);
     st_ctx->type = RX;
+    std::lock_guard<std::mutex> lock(ctx_mtx);
```

Comment:
mDpCtx is also used in api_server_tcp.cc line 133, 215 and 244
I think lock should be applied there too

Reply:
done